### PR TITLE
chore: update to Electron 19

### LINKED ISSUE #40262: Bottom left "action" (gear icon) menu not scrolling (MacOS)
- VSCode Version: Code 1.19.0 (816be6780ca8bd0ab80314e11478c48c70d09383, 2017-12-14T09:51:12.123Z)
- OS Version: Darwin x64 16.7.0
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
vscode-markdownlint|Dav|0.12.0
project-manager|ale|0.23.1
alignment|ann|0.3.0
code-gnu-global|aus|0.2.2
togglehs|bbe|0.1.2
gitignore|cod|0.5.0
githistory|don|0.2.3
git-project-manager|fel|1.4.0
auto-close-tag|for|0.5.5
Go|luk|0.6.69
vscode-clang|mit|0.2.2
cpptools|ms-|0.14.4
debugger-for-chrome|msj|4.0.0
vscode-emacs-improved|rkw|0.1.1
cmake|twx|0.0.17
vscode-lldb|vad|0.7.1
cmake-tools|vec|0.10.3
gitblame|wad|2.2.0
clang-format|xav|1.6.1

(7 theme extensions excluded)

---

Steps to Reproduce:

1. Resize VS Code window so it occupies most but not all, of the display.
2. Click on bottom left "action" (gear icon) menu because it has
   an indicator on it (I'd just updated and it was the restart button).
3. Menu draws as a drop down but is clipped due to lack of space. See arrow at bottom of menu indicating it can scroll.
4. Attempt to scroll menu clicking on arrow and usual MacOS pointer motion. Menu doesn't scroll.
5. Resize window to give more room below (effectively moving the gear icon up a little). Click gear icon and see full menu.

The menu display changes location at different window sizes, which really means available space for the menu's height.  With a slightly larger window size the menu renders itself above and to the right of the gear icon.  Reducing the window height a little has it render as a drop-down (below) but with  insufficient height to display the full menu and adds the non-functioning scroll arrow.

Reproduces without extensions:  Yes

### LINKED ISSUE #150475: no touch response for close/minimize/maximize vscode window controls
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version:
- OS Version:

I can't touch the window controls to close/minimize/maximize in the Insiders Build on W11.  Note that it works in Stable v1.67.2.

The vscode window controls are responsive to a mouse however.

And, interestingly, the Layout Control button is touch-responsive.

Version: 1.68.0-insider (user setup)
Commit: 528ee1ae3daabe30c1307cf9dcd6e77eb96094bc
Date: 2022-05-26T05:16:43.433Z
Electron: 17.4.4
Chromium: 98.0.4758.141
Node.js: 16.13.0
V8: 9.8.177.13-electron.0
OS: Windows_NT x64 10.0.25120

### LINKED ISSUE #150779: WCO: minimize button stays selected
Steps:
* minimize window
* restore window

=> 🐛 the minimize button shows selected

image

### LINKED ISSUE #153849: Segmentation fault opening vscode on Arch Linux ARM64
Does this issue occur when all extensions are disabled?: Yes

Hello Friends,

I'm running vscode from Arch Linux ARM (Asahi) on my MacBook Air M1 and its giving me a segfault before it opens.

`/opt/visual-studio-code/bin/code: line 62: 16644 Segmentation fault      ELECTRON_RUN_AS_NODE=1 "$ELECTRON" "$CLI" --ms-enable-electron-run-as-node "$@"`

I have never opened code before on this system and I think it may be related to the fact that the M1 uses a 16K address page (chromium had a [similar issue](https://bugs.chromium.org/p/chromium/issues/detail?id=1301788) and maybe a [similar patch](https://chromium.googlesource.com/chromium/src/+/a704c3a94179e6f3abb28f090d096ced72657d7c) can be applied) instead of a 4K. More info in the [Asahi wiki](https://github.com/AsahiLinux/docs/wiki/Broken-Software#if-package-supports-aarch64-why-doesnt-it-work).

I know its a very niche case on a probably unsupported platform but since more ARM systems would start using pages bigger than the standard x86 4K for the big performance improvement it may be worth it the report.

- VS Code Version: Both 1.68.1 and Insiders build 1656566876
- OS Version: Arch Linux Asahi ARM64 (MacBook Air M1) - Linux 5.17.0-rc7-asahi-next-20220310-5-2-ARCH

Steps to Reproduce:

1. Open visual studio code with command `/usr/bin/code --unity-launch %F` or simply `code`.

### LINKED ISSUE #154693: New version 1.69 has missing `app_id` under wayland
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.69.0
- OS Version: Linux wayland backend

Steps to Reproduce:
- Under any wayland desktop
- Force wayland backend for vscode
```
$ cat ~/.config/code-flags.conf 
--enable-features=UseOzonePlatform
--ozone-platform=wayland
```
- Any top level window under wayland is identified by `app_id` which used to be equal to "code" and now it is an empty string.

This bug is probably caused by switching to electron 18, https://github.com/electron/electron/issues/33578
 `app_id` is the only real way to differentiate windows under wayland if you are using a tiling window manager and want to pin some applications to some workspaces.

### LINKED ISSUE #154705: Highlighting, flickering, and rendering graphics display issues
Does this issue occur when all extensions are disabled?: **Yes**

- VS Code Version: 1.69.0
- OS Version: Fedora 36 (VMware guest VM)

Steps to Reproduce:

1. Upgrade from 1.68.1 to 1.69.0
2. Open a code file and start to try and highlight blocks of code with your mouse
3. After a few seconds of doing this repeated you will get strange flickering and blocking display issues

This appears to be a regression from 1.68.1 since didn't happen on my system with this version, and when I downgrade back to 1.68.1 the problem of course disappears, so doesn't appear to be caused by any very recent Fedora system package updates. In 1.69.0, if I launch with `--disable-gpu`, then the issue also disappears.

Screenshot from 2022-07-10 11-52-48

### PR DESCRIPTION
**Pull request by deepak1556**
- [x] Nightly exploration builds

Fixes https://github.com/microsoft/vscode/issues/154693
Fixes https://github.com/microsoft/vscode/issues/154705
Fixes https://github.com/microsoft/vscode/issues/153849
Fixes https://github.com/microsoft/vscode/issues/150779
Fixes https://github.com/microsoft/vscode/issues/40262
Fixes https://github.com/microsoft/vscode/issues/150475

### CODE REVIEW

**Comment by deepak1556** [comment]
@moham96 the fix is already available in VSCode stable `1.69` as part of Electron 18 update, you can refer to https://github.com/microsoft/vscode/pull/145527#issuecomment-1141146714 for context.

**Comment by moham96** [comment]
> @moham96 the fix is already available in VSCode stable `1.69` as part of Electron 18 update, you can refer to [#145527 (comment)](https://github.com/microsoft/vscode/pull/145527#issuecomment-1141146714) for context.

well, i tried using the [aur version](https://aur.archlinux.org/packages/visual-studio-code-bin) of visual-studio-code which pulls the 1.69.1 version but it still gives me segmentation fault, compiling the `electron-19.x.y` branch doesn't result in a segmentation fault

**Comment by ayr-ton** [comment]
Fixes #153849

**Comment by ReillyBrogan** [comment]
You should bump to 19.0.10 which includes the `--ozone-platform-hint=auto` flag, useful for those who want to switch between Wayland and X11 backends automatically on Linux.

**Comment by Austaras** [comment]
19.0.10 fixes electron/electron#33578 too.

File: .github/workflows/basic.yml [hunk 1]
```diff
@@ -39,8 +39,8 @@ jobs:
         uses: actions/cache@v3
         with:
           path: "**/node_modules"
-          key: ${{ runner.os }}-cacheNodeModules22-${{ steps.nodeModulesCacheKey.outputs.value }}
-          restore-keys: ${{ runner.os }}-cacheNodeModules22-
+          key: ${{ runner.os }}-cacheNodeModules23-${{ steps.nodeModulesCacheKey.outputs.value }}
```
Comment (bpasero) [inline .github/workflows/basic.yml:42]: If we continue to change this for each electron version then maybe this suffix should be more specific and have the electron major version in it?

**Review by bpasero — DISMISSED** [review]

File: .github/workflows/basic.yml (hunk 1 above)
Comment (deepak1556) [reply .github/workflows/basic.yml:42]: Yeah makes sense, will update the keys.

Comment (deepak1556) [reply .github/workflows/basic.yml]: Investigated a bit more before changing the key name as to why the computed key from https://github.com/microsoft/vscode/blob/main/build/azure-pipelines/common/computeNodeModulesCacheKey.ts is not sufficient in Github CI whereas it works well for azure pipelines when just updating the `.cachesalt`. Turns out we are using `restore-keys` option which can use stale cache when no cache hit was found https://github.com/actions/cache#inputs, I just removed it which should avoid bumping the key name with every Electron update.

File: build/linux/debian/dependencies-generator.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ const fs_1 = require("fs");
 const os_1 = require("os");
 const path = require("path");
 const dep_lists_1 = require("./dep-lists");
+const manifests = require("../../../cgmanifest.json");
```
Comment (deepak1556) [inline build/linux/debian/dependencies-generator.js:13]: /cc @rzhao271 fyi

Ensures that these files are in sync with Electron updates.

**Comment by rzhao271** [comment]
Looks like the libgcc dependency changed

**Comment by bpasero** [comment]
Should @rzhao271 review the other changes?

**Comment by bpasero** [comment]
My 👍 still stands, leaving new changes for Raymond.

**Review by rzhao271 — DISMISSED** [review]

**Review by bpasero — APPROVED** [review]

**Pull request merged**